examples/convert-dxil.rs

```diff
@@ -0,0 +1,166 @@
+use std::ffi::CStr;
+
+use saxaboom::{
+    IRComparisonFunction, IRCompilerFactory, IRFilter, IRMetalLibBinary, IRObject,
+    IRReflectionVersion, IRRootConstants, IRRootParameter1, IRRootParameter1_u,
+    IRRootParameterType, IRRootSignature, IRRootSignatureDescriptor1, IRRootSignatureFlags,
+    IRRootSignatureVersion, IRShaderReflection, IRShaderStage, IRShaderVisibility,
+    IRStaticBorderColor, IRStaticSamplerDescriptor, IRTextureAddressMode,
+    IRVersionedRootSignatureDescriptor, IRVersionedRootSignatureDescriptor_u,
+};
+
+fn create_static_sampler(
+    min_mag_mip_mode: IRFilter,
+    address_mode: IRTextureAddressMode,
+    index: u32,
+    anisotropy: Option<u32>,
+) -> IRStaticSamplerDescriptor {
+    let max_anisotropy = anisotropy.unwrap_or(1);
+
+    IRStaticSamplerDescriptor {
+        Filter: min_mag_mip_mode,
+        AddressU: address_mode,
+        AddressV: address_mode,
+        AddressW: address_mode,
+        MipLODBias: 0.0,
+        MaxAnisotropy: max_anisotropy,
+        ComparisonFunc: IRComparisonFunction::IRComparisonFunctionNever,
+        MinLOD: 0.0,
+        MaxLOD: 100000.0,
+        ShaderRegister: index,
+        RegisterSpace: 0,
+        ShaderVisibility: IRShaderVisibility::IRShaderVisibilityAll,
+        BorderColor: IRStaticBorderColor::IRStaticBorderColorTransparentBlack,
+    }
+}
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    unsafe {
+        // Create an instance of IRCompiler
+        let compiler_factory = IRCompilerFactory::new("libmetalirconverter.dylib").unwrap();
+        let mut compiler = compiler_factory.create_compiler();
+
+        // Create an explicit root signature layout
+        let mut parameters = create_root_parameters();
+        let mut static_samplers = create_static_samplers();
+
+        let desc_1_1 = IRRootSignatureDescriptor1 {
+            Flags: IRRootSignatureFlags::IRRootSignatureFlagCBVSRVUAVHeapDirectlyIndexed,
+            NumParameters: parameters.len() as u32,
+            pParameters: parameters.as_mut_ptr(),
+            NumStaticSamplers: static_samplers.len() as u32,
+            pStaticSamplers: static_samplers.as_mut_ptr(),
+        };
+
+        let desc = IRVersionedRootSignatureDescriptor {
+            version: IRRootSignatureVersion::IRRootSignatureVersion_1_1,
+            __bindgen_anon_1: IRVersionedRootSignatureDescriptor_u { desc_1_1 },
```

Comment:
Maybe we can rename the fields with changes from https://github.com/Traverse-Research/saxaboom/tree/switch-to-bindgen.

Reply:
Implemented in latest commit